src/common/snippets/src/lowered/pass/fuse_loops.cpp

```diff
@@ -24,40 +24,32 @@ using LoopInfoPtr = LoopManager::LoopInfoPtr;
+                                          const LinearIR::LoopManager::LoopInfoPtr& loop_lower) {
+    auto found_port = [](const std::vector<LoopManager::LoopPort>& loop_ports, const ExpressionPort& target_port) {
+        return std::find_if(loop_ports.cbegin(), loop_ports.cend(),
+                            [&target_port](const LoopManager::LoopPort& loop_port) {return *(loop_port.expr_port.get()) == target_port; });
+    };
+    const auto& upper_exit_ports = loop_upper->get_exit_points();
+    const auto& lower_entry_ports = loop_lower->get_entry_points();
+    for (const auto& lower_entry_port : lower_entry_ports) {
+        const auto& src_port = lower_entry_port.expr_port->get_port_connector_ptr()->get_source();
+        const auto upper_exit_port_it = found_port(upper_exit_ports, src_port);
+        if (upper_exit_port_it != upper_exit_ports.cend()) {
+            const auto& upper_exit_port = *upper_exit_port_it;
+            if (!lower_entry_port.is_incremented || !upper_exit_port.is_incremented)
                 return false;
-            auto src_loop_port = loop_manager->get_loop_port_by_expr_port(src_port, loop_upper_id);
-            if (!src_loop_port.is_incremented)
-                return false;
-            if (entry.dim_idx != src_loop_port.dim_idx)
+            if (lower_entry_port.dim_idx != upper_exit_port.dim_idx)
                 return false;
         }
     }
     return true;
 }
 
-bool FuseLoops::can_be_fused(const LoopInfoPtr& loop_current, const LoopInfoPtr& loop_target) {
-    auto current_work_amount = loop_current->get_work_amount();
-    auto target_work_amount = loop_target->get_work_amount();
-    // Loop fusion is supported only if Loops have equal increments and the equal/broadcastable work amounts.
-    // Note: For example, Broadcastable work amounts are possible in the following case:
-    //     Relu_0 [16x1]     Relu_1 [16x128]
-    //                \           /
-    //                 Add [16x128]
-    // Because of expression order in linear IR and work of MarkLoop algorithm, there are 2 Inner Loops:
-    //  - Relu_0 with work amount `1` and increment `vector size`
-    //  - Relu_1 and Add with work amount `128` and increment `vector size`
-    // We can fuse them into one Loop with work amount `128` and increment `vector size`
-    const auto supported_work_amount = current_work_amount == target_work_amount || current_work_amount == 1 || target_work_amount == 1;
-    const auto supported_increment = loop_current->get_increment() == loop_target->get_increment();
-    return supported_work_amount && supported_increment;
+bool FuseLoops::can_be_fused(const LinearIR::LoopManager::LoopInfoPtr& loop_upper, const LinearIR::LoopManager::LoopInfoPtr& loop_lower) {
+    if (!loop_ports_are_compatible(loop_upper, loop_lower))
+        return false;
+    return loop_upper->get_increment() == loop_lower->get_increment();
```

Comment:
Ok, but that's the point of this method - to check that the loops can be fused.
We can't assume that if the method was called then the loops have equal work amount. 
Regarding the dynamic WA - it's a solid point, but we can just check that both loops have dynamic work amounts, no need to eliminate the check entirely.

My concern here is that this check allows to fuse loops with equal increments, but different work amounts. We may not have such loops in IR right now, but they can be created in future. And the method should yield correct result for them.

Let's discuss this offline if needed.

Reply:
Seems like I got your point:
- If the both `work_amounts` are static - they must be broadcastable or equal (now it's missed in the method)
- If one of them is dynamic (or the both are dynamic) - we can fuse loops and set common `work_amount` as `dynamic`.
Thus, I've added check here for static case. I hope I got you correctly 😃 Otherwise correctly me please

Reply:
Exactly.
Could you please also explain why do you allow for broadcastable work amounts? `utils::one_of(1lu, work_amount_upper, work_amount_lower)`. I mean dimensions should be broadcastable, but work amounts?
My concern is that if we merge `loop 1` with `work_mount=1` and `loop 2` with `work_mount=1024`, we will have execute `loop 1`'s body 1024 times as a result, which will affect the overall performance.

Reply:
Seems like I lost commit with applying of your comments after rebasing 😄 
I added this commit https://github.com/openvinotoolkit/openvino/pull/21895/commits/ddd8856beb8653382ca97fdc20a801bba990d052 with previous comment that describes your case. Could you take a look please at the description? I'm sorry for the inconveniences!

Moreover, I noticed that before we had `increment=vector_size` for cases `work_amount=1` but now we have `work_amount=increment=1` (normalized increment). Because of that my previous condition was invalid in this PR. I returned Vladislav solution and update it to add dynamism support here.

Reply:
Ok. Looks better now, now more questions 🙂